### DISCUSSION TITLE
RTPS discovery over a wireguard VPN

### DISCUSSION THREAD

**Discussion by tmayoff** [OpenDDS/OpenDDS#4142] [Q&A]
I'm trying to get RTPS discovery working over a wireguard VPN, and I can't get it working. By default I believe RTPS discovery uses multicast and I've read conflicting things about whether multicast traffic can be sent over wireguard, I've tried setting up the allowed IPs for it but no luck. I tried adjusting the DDS discovery config, with various combinations of the options below, but also to no avail. `SedpMulticast=0` was the first thing I tried as the documentation says it disabled multicast, but with wireshark it looks like there's still multicast packets being sent. Any help is appreciated, thanks

```ini
[domain/2]
DefaultTransportConfig=vpnconfig_tcp
DiscoveryConfig=custom_rtps_config

[rtps_discovery/custom_rtps_config]
SedpMulticast=0
SpdpLocalAddress=10.0.7.2
SedpAdvertisedLocalAddress=10.0.7.2:
MulticastInterface=wg0

[transport/vpntransport_tcp]
transport_type=tcp
local_address=10.0.7.2:
```

**Comment by mitza-oci** [comment]
`SedpMulticast=0` disables multicast for SEDP, not for discovery altogether.  You can configure discovery to send SPDP announcements to any address (unicast or multicast).

**Reply by tmayoff** [reply]
What's the config option to do that, `SpdpLocalAddress` doesn't seem to work, nothing connects.

**Reply by jrw972** [reply]
See `SpdpSendAddrs` https://opendds.readthedocs.io/en/latest-release/devguide/run_time_configuration.html
